tabular/src/autogluon/tabular/models/image_prediction/image_predictor.py

```diff
@@ -127,9 +146,44 @@ def _fit(self,
 
     def _predict_proba(self, X, **kwargs):
         X = self.preprocess(X, **kwargs)
-        y_pred_proba = self.model.predict_proba(X, as_pandas=False)
+        # TODO: Add option to crash if null is present for faster predict_proba
+        null_indices = X['image'] == ''
+        if null_indices.sum() > 0:
+            if self.num_classes is None:
+                y_pred_proba = np.zeros(len(X))
+            else:
+                y_pred_proba = np.zeros((len(X), self.num_classes))
+            X = X.reset_index(drop=True)
+            null_indices = X['image'] == ''
+            X = X[~null_indices]
+            null_indices_rows = list(null_indices[null_indices].index)
+            non_null_indices_rows = list(null_indices[~null_indices].index)
+            y_pred_proba[null_indices_rows] = self._dummy_pred_proba
+            y_pred_proba[non_null_indices_rows] = self.model.predict_proba(X, as_pandas=False)
+        else:
+            y_pred_proba = self.model.predict_proba(X, as_pandas=False)
         return self._convert_proba_to_unified_form(y_pred_proba)
 
+    # TODO: Consider moving to AbstractModel or as a separate function
+    # TODO: Test softclass
+    def _compute_dummy_pred_proba(self, y):
+        num_classes = self.num_classes
+        if self.problem_type in [BINARY, MULTICLASS]:
+            dummies = pd.get_dummies(y)
+            dummy_columns = set(list(dummies.columns))
+            if len(dummies.columns) < num_classes:
+                for c in range(num_classes):
+                    if c not in dummy_columns:
+                        dummies[c] = 0
+            dummies = dummies[list(range(num_classes))]
```

Comment:
dummies = dummies.iloc[range(num_classes)]

Reply:
That isn't what I am trying to do, I am re-ordering the column names so it ensures they are in the same order as in predict_proba output.